### PR TITLE
Add functionality to extract formatted cells from a gt table object

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `extract_cells()` function and allows for extraction of a vector of formatted cells from a gt table.

Fixes: https://github.com/rstudio/gt/issues/1019

### CODE REVIEW

File: R/dt_formats.R [hunk 1]
```diff
@@ -12,15 +12,11 @@ dt_formats_init <- function(data) {
   dt_formats_set(data = data, formats = list())
 }
 
-dt_formats_add <- function(data, formats, prepend) {
```
Comment (cscheid) [inline R/dt_formats.R:15]: What happens with user code that depends on `prepend`?

**Review by cscheid — APPROVED** [review]
Assuming my `dt_formats` question has an answer that doesn't require action, this lgtm.

File: R/dt_formats.R (hunk 1 above)
Comment (rich-iannone) [reply R/dt_formats.R:15]: It never worked anyway. This was a vexing bug but after unraveling it we now have something stable and better tested.

Comment (cscheid) [reply R/dt_formats.R:15]: Ok, sounds good!

**Pull request merged**